zkevm-circuits/src/ecc_circuit.rs

```diff
@@ -438,24 +471,12 @@ impl<F: Field, const XI_0: i64> EccCircuit<F, XI_0> {
         let rand_point = ecc_chip.load_random_point::<G1Affine>(ctx);
 
         // check if P == (0, 0), Q == (0, 0), R == (0, 0)
-        let p_x_is_zero = ecc_chip
-            .field_chip
-            .is_zero(ctx, &point_p.decomposed.ec_point.x);
-        let p_y_is_zero = ecc_chip
-            .field_chip
-            .is_zero(ctx, &point_p.decomposed.ec_point.y);
-        let q_x_is_zero = ecc_chip
-            .field_chip
-            .is_zero(ctx, &point_q.decomposed.ec_point.x);
-        let q_y_is_zero = ecc_chip
-            .field_chip
-            .is_zero(ctx, &point_q.decomposed.ec_point.y);
-        let r_x_is_zero = ecc_chip
-            .field_chip
-            .is_zero(ctx, &point_r.decomposed.ec_point.x);
-        let r_y_is_zero = ecc_chip
-            .field_chip
-            .is_zero(ctx, &point_r.decomposed.ec_point.y);
+        let p_x_is_zero = ecc_chip.field_chip.is_zero(ctx, &point_p.ec_point.x);
```

Comment:
I realized that those constraints are not required. 

During `load_random_point()` it is already checked that the point is on curve, i.e., `y^2 = x^3 + b`. For point `(0,0)` it will not pass `load_random_point()`

In other words, the `load_random_point()` cannot load an infinity point

Reply:
The check for `(0, 0)` is not for `rand_point` (that was loaded at random), but for `P`, `Q` and `R` (those are the points from EVM inputs to `ecAdd`.

Reply:
oh I see... then there maybe another issue: have we checked that `P`, `Q` and `R` are on curve?

Reply:
oh ok. This is missing. I hadn't looked in-depth into the `halo2-ecc`'s `load_private` method. It does not check if the point is on the curve.

I can refactor it in a way which does this:
- assert `is_on_curve_or_zero` and return `is_zero`
- use the `is_zero` to form the rest of the constraints

Wdyt?